test/unit/supervisor_test.rb

```diff
@@ -185,6 +185,33 @@ class SupervisorTest < ActiveSupport::TestCase
     end
   end
 
+  # Regression test for supervisor failing to handle claimed jobs when its own
+  # process record has been pruned (NoMethodError in #handle_claimed_jobs_by).
+  test "handle_claimed_jobs_by fails claimed executions even if supervisor record is missing" do
+    worker_name = "worker-test-#{SecureRandom.hex(4)}"
+
+    worker_process = SolidQueue::Process.register(kind: "Worker", pid: 999_999, name: worker_name)
+
+    job = StoreResultJob.perform_later(42)
+    claimed_execution = SolidQueue::ReadyExecution.claim("*", 1, worker_process.id).first
+
+    terminated_fork = Struct.new(:name).new(worker_name)
+
+    DummyStatus = Struct.new(:pid, :exitstatus) do
+      def signaled? = false
+      def termsig = nil
+    end
+    status = DummyStatus.new(worker_process.pid, 1)
+
+    supervisor = SolidQueue::Supervisor.allocate
+
+    supervisor.send(:handle_claimed_jobs_by, terminated_fork, status)
```

Comment:
This test is terribly synthetic, I'm afraid. Just explicitly constructs the situation that's encountered in the wild (nil Supervisor#process) rather than realistically (actually pruning the process).